addons/base_automation/models/base_automation.py

```diff
@@ -742,7 +743,7 @@ def write(self, vals, **kw):
                 pre = {a: a._filter_pre(records) for a in automations}
                 # read old values before the update
                 old_values = {
-                    record.id: {field_name: record[field_name] for field_name in vals}
+                    record.id: {field_name: record[field_name] for field_name in vals if not is_selection_groups(field_name)}
```

Comment:
`if not is_selection_groups(field_name)` is specific to res.users. base.automation shouldn't contain specific logic of other models IMO. I'd rather make something generic to accept any kind of weird write override. How about  `if field_name in record._fields` ? This also has the advantage of letting `write` raise the correct [ValueError](https://github.com/odoo/odoo/blob/6b907f92076271f6aec5239bc0d6d9f1ab7986b1/odoo/models.py#L4523).

Reply:
Amended and tested with this change. Let me know if you have any other feedback!